detection.go

```diff
@@ -71,10 +71,41 @@ func isAOneOfField(v reflect.Value, sf reflect.StructField) bool {
 	return v.Kind() == reflect.Interface && sf.Tag.Get("protobuf_oneof") != ""
 }
 
+// isAProto2BytesField checks if the field is a proto2 bytes field.
+//
+// This is done by checking the field's tag. Byte fields do not have a 'rep'
+// tag (for repeated fields).  Additionally, proto3 byte fields have a 'proto3'
+// tag, which proto2 byte fields do not have.
+func isAProto2BytesField(v reflect.Value, sf reflect.StructField) bool {
+	tag := sf.Tag.Get("protobuf")
+
+	// This can potentially break "def" values which come last and can contain
+	// commas. This is fine because we don't care about "def" or its value here.
+	//
+	// Example: "bytes,10,opt,name=F_String,json=FString,def=hello, \"world!\"\n"
+	fields := strings.Split(tag, ",")
+
+	// Normal proto tags (incl. bytes) will have at least 2 fields.
+	if len(fields) < 2 {
+		// We do not return an error because this case can happen for certain kinds
+		// of fields, such as 'XXX_unrecognized' for example.
+		return false
+	}
+
+	for i := 2; i < len(fields); i++ {
+		f := fields[i]
+		if f == "rep" || f == "proto3" {
```

Comment:
Where are you checking that its a bytes field at all?

Reply:
See lines 76-78 above:

```
// This is done by checking the field's tag. Byte fields do not have a 'rep'
// tag (for repeated fields).  Additionally, proto3 byte fields have a 'proto3'
// tag, which proto2 byte fields do not have.
```

There's no direct way to test for something being a bytes field, so I check that it's not a repeated field to know it's a bytes field. There's also no direct way to test for a bytes field being proto2, but all proto3 bytes fields have a "proto3" tag :stuck_out_tongue:

Reply:
My comment was more about what happens if you call this with something like an int type.  You're assuming the caller calls it with a proper type.   Comment or check on expected input type would be good.

Reply:
Done